### DISCUSSION TITLE
Migration to make column nullable requires DB reset?

### DISCUSSION THREAD

**Discussion by sauntimo** [prisma/prisma#11307] [Q&A]
I changed my schema in DBeaver, dropping a `NOT NULL` constraint on three columns in one table, and then ran the following commands 

```shell
prisma db pull
prisma generate
and prisma migrate dev --name billing-props-nullable
```
and was greeted with
```
? We need to reset the PostgreSQL database "postgres" at "[my postgres url]".
Do you want to continue? All data will be lost. › (y/N)
```
This operation should be safe to carry out. If I was adding a `NOT NULL` constraint, or something more complex, I could understand why truncating all the ables would make that easier. But in this case, I don't understand why prisma wants to nuke my database in order to create this migration? It's not like it's even deploying anything, I've already made the change on my development database.

I've also run into this when I was simply changing the name of a constraint which also strikes me as weird. Do you always have to delete all of your data for every single migration? Surely that can't be right. Am I running the wrong commands?

At this stage in development I don't yet have any data in this database (I'm transitioning from firebase) but when I've fully switched to supabase, I don't want prisma to drop my data when I make a schema change - and I feel like there shouldn't be any need to - so I feel like I must have misunderstood the process?

**Comment by austincrim** [comment]
Hey @sauntimo 👋 ,

Could you run `prisma migrate dev --name billing-props-nullable --create-only` and provide me with the generated SQL?

Thanks so much!

**Reply by sauntimo** [reply]
Hey @austincrim, sorry I hadn't got back to you sooner - I've just run into this issue again so this is a good time to pick it back up. I'm adding a column to a table - should be perfectly safe! - but, prisma wants to nuke my db for reasons I don't understand.

Here's a screen shot of the messages I've run into
![image](https://user-images.githubusercontent.com/2720466/153911302-6759a046-299b-446b-8296-7c68510ff057.png)

I added the `--create-only` as you asked and it still wanted to destroy all my data. I've allowed it, and this is the SQL it has generated:
```sql
-- AlterTable
ALTER TABLE "user_mood_reaction" ADD COLUMN     "created_at" TIMESTAMPTZ(6) NOT NULL DEFAULT CURRENT_TIMESTAMP;
```
As you can see, there's nothing there that should require destruction of my data.

I wonder if the flow I am following is the problem? I'm making schema changes in DBeaver and then `prisma db pull`-ing them into prisma - that should be okay, shouldn't it?

**Reply by pantharshit00** [reply]
Hello @sauntimo, our apologies for the delay in getting back to you! Due to the large number of questions in our community channels, unfortunately this slipped through the cracks. It's very important for us to address your question, so let us know in a comment if this is still an open discussion. 

Any change you make outside of migrate will be considered a drift from the migrations history. So in order to resolve this, you will need to create a new migration and mark it as already applied so that migrate doesn't try to apply them. So can do so by doing the following:
```
mkdir -p prisma/migrations/remove-unique-constraint
npx prisma migrate diff --from-migrations prisma/migrations --to-schema-datasource prisma/schema.prisma --shadow-database-url <any-empty-db-url> --script > prisma/migrations/remove-unique-constraint/migration.sql
npx prisma migrate resolve --applied remove-unique-constraint
npx prisma migrate dev # it should say everything is in sync
```

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
